src/triage/predictlist/__init__.py

```diff
@@ -381,31 +311,77 @@ def get_feature_dict_and_imputation_task(self, collate_aggregations, model_id):
                 )
             )
         return reconstructed_feature_dict, imputation_table_tasks
-    
+
     def retrain(self, prediction_date):
         """Retrain a model by going back one split from prediction_date, so the as_of_date for training would be (prediction_date - training_label_timespan)
         
         Args:
             prediction_date(str) 
         """
+
+        retrain_config = {
+            "model_group_id": self.model_group_id,
+            "prediction_date": prediction_date,
+        }
+        self.retrain_hash = save_retrain_and_get_hash(retrain_config, self.db_engine)
+
+        with get_for_update(self.db_engine, Retrain, self.retrain_hash) as retrain:
+            retrain.prediction_date = prediction_date
```

Comment:
Is the idea here that we don't want the prediction date to be part of the retrain hash? But then if we re-run the same retrain and predict config with a different prediction date, won't we end up overwriting the date in the table by doing this?

Reply:
I think the prediction date is part of the retrain hash. The retrain_config has the prediction date and the retrain hash is generated by  `save_retrain_and_get_hash`.